File: rb109/Practice/Easy 9/welcome_stranger.rb
# welcome_stranger.rb

# write a method
# input: two arguments: an array and a hash
# array: contains 2+ elements that, combined with adjoining spaces, will
# produce a person's name;
# hash: contains two keys, :title and :occupation, and the appropriate values

# output: a string greeting that uses the person's full name, and mentions
# the person's title

def greetings(array, hash)
  puts "Hello, #{array.join(' ')}! Nice to have a #{hash[:title]} " \
       "#{hash[:occupation]} around."
end

greetings(['John', 'Q', 'Doe'], { title: 'Master', occupation: 'Plumber' })

File: js210/lesson_5_objects/welcome_stranger.js
function greetings(names, specs) {
  let name = names.join(' ');
  let title = `${specs.title} ${specs.occupation}`;
  let text = `Hello, ${name}! Nice to have a ${title} around.`;

  console.log(text);
}

greetings(['John', 'Q', 'Doe'], { title: 'Master', occupation: 'Plumber' });
